src/models/brightcove.js

```diff
@@ -155,22 +211,112 @@ class Brightcove extends Video {
 				if (titleEl) {
 					this.containerEl.removeChild(titleEl);
 				}
-				this.removePlaceholder();
-				this.addVideo();
+				this.containerEl.removeChild(this.placeholderEl);
+
+				this.el.style.display = 'block';
 				this.el.play();
 				this.el.focus();
 			});
 		}
-	}
 
-	removePlaceholder() {
-		this.containerEl.classList.remove('o-video--placeholder');
-		this.containerEl.removeChild(this.placeholderEl);
+		// Adds video soon so ads can start loading
+		this.addVideo();
+		// Hide it so it doesn't flash until the placeholder image loads
+		this.el.style.display = 'none';
 	}
 
 	getProgress() {
 		return this.el.duration ? parseInt(100 * this.el.currentTime / this.el.duration, 10) : 0;
 	}
+
+	playAdEventHandler() {
+		// Sets the styling now so the ad occupies the space of the video
+		this.adContainerEl.classList.add('o-video__ad');
+		// Initialize the video. Must be done via a user action on mobile devices.
+		this.el.load();
+		this.adDisplayContainer.initialize();
+
+		try {
+			// Initialize the ads manager. Ad rules playlist will start at this time.
+			this.adsManager.init(640, 360, google.ima.ViewMode.NORMAL);
```

Comment:
Give me a minute as I´m fixing the CI stuff and there might be conflicts

On Thu, 9 Jun 2016 at 12:19 Vlad Dubrovskis <notifications@github.com>
wrote:

> In src/models/brightcove.js
> <https://github.com/Financial-Times/o-video/pull/8#discussion_r66423173>:
>
> >  	}
> >
> >  	getProgress() {
> >  		return this.el.duration ? parseInt(100 * this.el.currentTime / this.el.duration, 10) : 0;
> >  	}
> > +
> > +	playAdEventHandler() {
> > +		// Sets the styling now so the ad occupies the space of the video
> > +		this.adContainerEl.classList.add('o-video__ad');
> > +		// Initialize the video. Must be done via a user action on mobile devices.
> > +		this.el.load();
> > +		this.adDisplayContainer.initialize();
> > +
> > +		try {
> > +			// Initialize the ads manager. Ad rules playlist will start at this time.
> > +			this.adsManager.init(640, 360, google.ima.ViewMode.NORMAL);
>
> i would change another value, let me make a comment on relevant line. or I
> can push a commit? :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Financial-Times/o-video/pull/8/files/a485e88fe0007ad7c8348119c55de8593e26f011#r66423173>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA3u3enbVWO6fjnEV53Z8Fm1XDQI3fxnks5qJ_bQgaJpZM4Ix1Qf>
> .
>

-- 

------------------------------

*This email was sent by a company owned by Financial Times Group Limited 
("FT Group <http://aboutus.ft.com/corporate-information/#axzz3rajCSIAt>"), 
registered office at Number One Southwark Bridge, London SE1 9HL.  
Registered in England and Wales with company number 879531. This e-mail may 
contain confidential information. If you are not the intended recipient, 
please notify the sender immediately, delete all copies and do not 
distribute it further.  It could also contain personal views which are not 
necessarily those of the FT Group.  We may monitor outgoing or 
incoming emails as permitted by law.*